File: emarsys/src/main/java/com/emarsys/service/EmarsysMessagingService.java
package com.emarsys.service;

import com.emarsys.Emarsys;
import com.google.firebase.messaging.FirebaseMessagingService;
import com.google.firebase.messaging.RemoteMessage;

public class EmarsysMessagingService extends FirebaseMessagingService {

    @Override
    public void onNewToken(String token) {
        super.onNewToken(token);
        Emarsys.Push.setPushToken(token);
    }

    @Override
    public void onMessageReceived(RemoteMessage remoteMessage) {
        super.onMessageReceived(remoteMessage);

        EmarsysMessagingServiceUtils.handleMessage(this, remoteMessage);
    }
}

File: emarsys-firebase/src/main/java/com/emarsys/service/EmarsysMessagingService.kt
package com.emarsys.service

import com.emarsys.mobileengage.di.mobileEngage
import com.google.firebase.messaging.FirebaseMessagingService
import com.google.firebase.messaging.RemoteMessage

class EmarsysMessagingService : FirebaseMessagingService() {
    override fun onNewToken(token: String) {
        super.onNewToken(token)
        mobileEngage().coreSdkHandler.post {
            if (mobileEngage().deviceInfo.isAutomaticPushSendingEnabled) {
                mobileEngage().pushInternal.setPushToken(token, null)
            }
        }
    }

    override fun onMessageReceived(remoteMessage: RemoteMessage) {
        super.onMessageReceived(remoteMessage)
        EmarsysMessagingServiceUtils.handleMessage(this@EmarsysMessagingService, remoteMessage)
    }
}